tests/test_utils.h

```diff
@@ -5,6 +5,20 @@
 using namespace std;
 using namespace fplll;
 
+#define read_file(X, input_filename) {\
```

Comment:
Why not make this a function instead of a define?

Reply:
Yes, I agree that it is (a bit) ugly. In fact, the function `read_file` reads `input_filename` and processes the result into `X`. Since `X` is either a `vector<Z_NR<ZT>>` or a `ZZ_mat<ZT>`, I only see this way to unify how to read something from a file and store it without specifying the type of `X`. A solution to avoid the `# define` is maybe to return the `ifstream` and test if it is empty or not.

Reply:
Couldn't we template it?

Reply:
Commit bdfe597 is supposed to do that.